binding.gyp

```diff
@@ -6,29 +6,37 @@
         [ "OS == 'win'", {}],
         [ "OS == 'linux'", {
           "libraries": [
-            "../deltachat-core/builddir/src/libdeltachat.a",
-            "../deltachat-core/builddir/libs/libetpan/libetpan.a",
-            "../deltachat-core/builddir/libs/netpgp/libnetpgp.a",
-            "-lsasl2",
-            "-lssl",
-            "-lsqlite3",
             "-lpthread"
           ],
           "cflags": [
-            "-std=gnu99"
-          ]
+            "-std=gnu99",
+          ],
+          "conditions": [
+            [ "'<!(echo $DC_SRC)x' == 'x'", {
+              "libraries": [
+                "<!(pkg-config --libs deltachat)",
+              ],
+              "cflags": [
+                "<!(pkg-config --cflags deltachat)"
+              ],
+            }],
+            [ "'<!(echo $DC_SRC)x' != 'x'", {
+              "libraries": [
+                "-L../<!(echo $DC_BUILD/src)",
+                "-ldeltachat",
+              ],
+              "cflags": [
+                "-I../<!(echo $DC_SRC/src)",
+              ],
+            }],
+          ],
         }],
         [ "OS == 'mac'", {
           "libraries": [
-            "../deltachat-core/builddir/src/libdeltachat.a",
-            "/usr/local/Cellar/libetpan/1.9.2_1/lib/libetpan.a",
-            "../deltachat-core/builddir/libs/netpgp/libnetpgp.a",
-            "-framework CoreFoundation",
-            "-framework CoreServices",
-            "-framework Security",
-            "-lsasl2",
-            "-lssl",
-            "-lsqlite3",
+            "<!(pkg-config --libs deltachat)",
+            # "-framework CoreFoundation",
+            # "-framework CoreServices",
+            # "-framework Security",
```

Comment:
Is it possible that we still need the -framework things on mac? Or is it enough if its in the libdeltachat? Did you test that? Travis is currently quite unhelpful ^^

Reply:
I'm still working out how osx works. But basically I expect that if these flags are needed they should be part of deltachat-core rather then here since the node bindings only call functions into dc-core and no other libraries.  So it should only have to care about linking to dc-core.